FSI/Stress.C

```diff
@@ -49,96 +32,17 @@ solverType_(solverType)
 {
-    // Compute stress. See the Forces function object.
-
-    // Stress tensor boundary field
-    tmp<volSymmTensorField> tdevRhoReff(this->devRhoReff());
-    const volSymmTensorField::Boundary& devRhoReffb
-    (
-        tdevRhoReff().boundaryField()
-    );
-
-    // Density boundary field
-    tmp<volScalarField> trho(this->rho());
-    const volScalarField::Boundary& rhob
-    (
-        trho().boundaryField()
-    );
-
-    // Pressure boundary field
-    tmp<volScalarField> tp(mesh_.lookupObject<volScalarField>("p"));
-    const volScalarField::Boundary& pb
-    (
-        tp().boundaryField()
-    );
-
-    int bufferIndex = 0;
-    // For every boundary patch of the interface
-    for (uint j = 0; j < patchIDs_.size(); j++)
-    {
-
-        int patchID = patchIDs_.at(j);
-
-        // Compute normal vectors on each patch
-        const vectorField nV = mesh_.boundary()[patchID].nf();
-
-        // Pressure constribution
-        if (solverType_.compare("incompressible") == 0)
-        {
-            Stress_->boundaryFieldRef()[patchID] =
-                nV * pb[patchID] * rhob[patchID];
-        }
-        else if (solverType_.compare("compressible") == 0)
-        {
-            Stress_->boundaryFieldRef()[patchID] =
-                nV * pb[patchID];
-        }
-        else
-        {
-            FatalErrorInFunction
-                << "Stress calculation does only support "
-                << "compressible or incompressible solver type."
-                << exit(FatalError);
-        }
-
-        // Viscous contribution
-        Stress_->boundaryFieldRef()[patchID] +=
-            nV & devRhoReffb[patchID];
-
-        // Write the stress to the preCICE buffer
-        // For every cell of the patch
-        forAll(Stress_->boundaryField()[patchID], i)
-        {
-            // Copy the stress into the buffer
-            // x-dimension
-            buffer[bufferIndex++]
-            = 
-            Stress_->boundaryField()[patchID][i].x();
-
-            // y-dimension
-            buffer[bufferIndex++]
-            =
-            Stress_->boundaryField()[patchID][i].y();
-
-            if(dim == 3)
-                // z-dimension
-                buffer[bufferIndex++]
-                        =
-                        Stress_->boundaryField()[patchID][i].z();
-        }
-    }
+  this->writeToBuffer(buffer, *Stress_, dim);
 }
 
 void preciceAdapter::FSI::Stress::read(double * buffer, const unsigned int dim)
 {
-    /* TODO: Implement
-    * We need two nested for-loops for each patch,
-    * the outer for the locations and the inner for the dimensions.
-    * See the preCICE readBlockVectorData() implementation.
-    */
-    FatalErrorInFunction
-        << "Reading stresses is not supported."
-        << exit(FatalError);
+    this->readFromBuffer(buffer);
+}
+
+vectorField preciceAdapter::FSI::Stress::getFaceVectors(const unsigned int patchID) const
+{
+  return mesh_.boundary()[patchID].nf();
```

Comment:
Maybe add a comment here on what `nf()` does, similar to the one for `Sf()`?

Reply:
Done.